apps/binaryclk/app.js

```diff
@@ -53,11 +59,47 @@ function draw() {
 		g.clearRect(Math.floor(mgn/2), mgn, Math.floor(mgn/2) + pos, mgn + c1sqhide * pos);
 		g.clearRect(Math.floor(mgn/2) + 2 * pos + gap, mgn, Math.floor(mgn/2) + 3 * pos, mgn + c3sqhide * pos);
 	}
+
 	if (settings.showdate) {
-		g.setFontAlign(0, 0);
+		g.setColor(-1).fillRect(Math.floor(mgn/2) + gap, mgn + gap, Math.floor(mgn/2) + gap + sq, mgn + gap + sq);
+		g.setColor('#FF0000').fillRect(Math.floor(mgn/2) + gap, mgn + gap, Math.floor(mgn/2) + gap + sq, mgn + gap + 12);
+		g.setFontAlign(0, -1);
+		g.setFont("Vector",12);
+		g.setColor(-1).drawString(mon, Math.ceil(mgn/2) + gap + Math.ceil(sq/2) + 1, mgn + gap + 1);
+		g.setFontAlign(0, 1);
 		g.setFont("Vector",20);
+		g.setColor(1).drawString(d, Math.ceil(mgn/2) + gap + Math.ceil(sq/2) + 1, mgn + gap + sq + 2);
+		if (g.theme.dark) {
+			g.setColor(-1);
+		} else {
+			g.setColor(1);
+			g.drawLine(Math.floor(mgn/2) + gap, mgn + gap + 13, Math.floor(mgn/2) + gap + sq, mgn + gap + 13);
+		}
 		g.drawRect(Math.floor(mgn/2) + gap, mgn + gap, Math.floor(mgn/2) + gap + sq, mgn + gap + sq);
-		g.drawString(d, Math.ceil(mgn/2) + gap + Math.ceil(sq/2) + 1, mgn + gap + Math.ceil(sq/2) + 1);
+	}
+
+	if (cnt == 0) {
+		if (settings.showbat) {
+			var bat = E.getBattery();
+			if (bat < 20) {
+				g.setColor('#FF0000');
+			} else if (bat < 40) {
+				g.setColor('#FFA500');
+			} else {
+				g.setColor('#00FF00');
+			}
+			g.fillRect(Math.floor(mgn/2) + gap + 2 * pos, mgn + gap, Math.floor(mgn/2) + gap + 2 * pos + Math.floor(bat * sq / 100), mgn + gap + sq);
+			if (g.theme.dark) {
+				g.setColor(-1);
+			} else {
+				g.setColor(1);
+			}
+			g.drawRect(Math.floor(mgn/2) + gap + 2 * pos, mgn + gap, Math.floor(mgn/2) + gap + 2 * pos + sq, mgn + gap + sq);
+		}
+		cnt =+ cnt;
```

Comment:
Did you mean something like this here?
```suggestion
		cnt += cnt;
```

or:
```suggestion
		cnt++;
```

Reply:
Should be updated now, thank you!